plugins/catalog-backend/src/service/router.ts

```diff
@@ -133,6 +150,11 @@ export async function createRouter(
     router.post('/locations', async (req, res) => {
       const input = await validateRequestBody(req, locationSpecSchema);
       const dryRun = yn(req.query.dryRun, { default: false });
+
+      // when in dryRun addLocation is effectively a read operation so when in
+      // dryRun we override mode to readwrite to allow the operation
+      requireReadWriteMode(dryRun ? 'readwrite' : mode);
```

Comment:
Prolly make this `if (!dryRun) { ... }` instead

Reply:
Good point 👍  Will update.